lib/image-stream/utils.js

```diff
@@ -17,6 +17,41 @@
 'use strict';
 
 const Bluebird = require('bluebird');
+const fs = Bluebird.promisifyAll(require('fs'));
+const errors = require('../shared/errors');
+
+/**
+ * @summary Read a buffer from an image file descriptor
+ * @function
+ * @private
+ *
+ * @param {Number} fileDescriptor - file descriptor
+ * @param {Number} count - number of bytes to read
+ * @param {Number} offset - image offset to start reading from
+ * @fulfil {Buffer} - buffer
+ * @returns {Promise}
+ *
+ * @example
+ * fs.openAsync('path/to/image.img', 'r').then((fileDescriptor) => {
+ *   return utils.readBufferFromImageFileDescriptor(fileDescriptor, 512, 0);
+ * }).then((buffer) => {
+ *   console.log(buffer);
+ * });
+ */
+exports.readBufferFromImageFileDescriptor = (fileDescriptor, count, offset) => {
```

Comment:
Can we swap offset & count here to `(fd, position, length)`, to adhere to the function signatures in the rest of Node's API?